cpp/CMakeLists.txt

```diff
@@ -465,11 +520,17 @@ if(NOT BUILD_CPU_ONLY)
     PUBLIC rmm::rmm raft::raft ${CUVS_CTK_MATH_DEPENDENCIES}
     PRIVATE nvidia::cutlass::cutlass $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX> cuvs-cagra-search
   )
+
+  target_link_libraries(
+    cuvs_static
+    PUBLIC rmm::rmm raft::raft ${CUVS_CTK_MATH_DEPENDENCIES}
+    PRIVATE nvidia::cutlass::cutlass $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX>
```

Comment:
Shouldn't this target also link `cuvs-cagra-search`?

Reply:
yeah - I agree this should link cuvs-cagra-search (and tried in the previous commit) .   The error I was hitting was : 

```
  CMake Error in CMakeLists.txt:
    export called with target "cuvs_static" which requires target
    "cuvs-cagra-search" that is not in any export set.
```

https://github.com/rapidsai/cuvs/actions/runs/11155746120/job/31007185860

Will try your suggestion - thanks for this!

Reply:
works great - thanks! applied in latest commit

Reply:
Do you think we still need the `cuvs-cagra-search` in the install export set? Theoretically, we shouldn't be exposing that as a library and nobody outside cuVS should need to link against it.